dlpy/network.py

```diff
@@ -0,0 +1,1598 @@
+            elif layer_type == 10:
+                self.layers.append(extract_concatenate_layer(layer_table=layer_table))
+            elif layer_type == 11:
+                self.layers.append(extract_detection_layer(layer_table=layer_table))
+
+        conn_mat = model_table[['_DLNumVal_', '_DLLayerID_']][
+            model_table['_DLKey1_'].str.contains('srclayers')].sort_values('_DLLayerID_')
+        layer_id_list = conn_mat['_DLLayerID_'].tolist()
+        src_layer_id_list = conn_mat['_DLNumVal_'].tolist()
+
+        for row_id in range(conn_mat.shape[0]):
+            layer_id = int(layer_id_list[row_id])
+            src_layer_id = int(src_layer_id_list[row_id])
+            if self.layers[layer_id].src_layers is None:
+                self.layers[layer_id].src_layers = [self.layers[src_layer_id]]
+            else:
+                self.layers[layer_id].src_layers.append(self.layers[src_layer_id])
+
+        # Check if weight table is in the same path
+        _file_name_, _extension_ = os.path.splitext(file_name)
+
+        _file_name_list_ = list(self._retrieve_('table.fileinfo',
+                                                caslib=cas_lib_name,
+                                                includeDirectories=False).FileInfo.Name)
+
+        if (_file_name_ + '_weights' + _extension_) in _file_name_list_:
+            print('NOTE: ' + _file_name_ + '_weights' + _extension_ +
+                  ' is used as model weigths.')
+
+            self._retrieve_('table.loadtable',
+                            caslib=cas_lib_name,
+                            path=_file_name_ + '_weights' + _extension_,
+                            casout=dict(replace=True, name=self.model_name + '_weights'))
+            self.set_weights(self.model_name + '_weights')
+
+            if (_file_name_ + '_weights_attr' + _extension_) in _file_name_list_:
+                print('NOTE: ' + _file_name_ + '_weights_attr' + _extension_ +
+                      ' is used as weigths attribute.')
+                self._retrieve_('table.loadtable',
+                                caslib=cas_lib_name,
+                                path=_file_name_ + '_weights_attr' + _extension_,
+                                casout=dict(replace=True,
+                                            name=self.model_name + '_weights_attr'))
+                self.set_weights_attr(self.model_name + '_weights_attr')
+        #if not flag:
+        #    self._retrieve_('table.dropcaslib', caslib=cas_lib_name)
+
+    def load_weights(self, path, labels=False):
+        '''
+        Load the weights form a data file specified by ‘path’
+
+        Parameters
+        ----------
+        path : string
+            Specifies the server-side directory of the file that
+            contains the weight table.
+
+        Notes
+        -----
+        Currently support HDF5 and sashdat files.
+
+        '''
+
+        dir_name, file_name = os.path.split(path)
```

Comment:
is the path server-side or client-side? if server-side we need to check the os of the server to split